osc/gitea_api/pr.py

```diff
@@ -27,52 +31,149 @@ def split_id(cls, pr_id: str) -> Tuple[str, str, int]:
-            return "yes" if value else "no"
+    @property
+    def id(self) -> str:
+        return f"{self.base_owner}/{self.base_repo}#{self.number}"
+
+    @property
+    def number(self) -> int:
+        return self._data["number"]
+
+    @property
+    def title(self) -> str:
+        return self._data["title"]
+
+    @property
+    def body(self) -> str:
+        return self._data["body"]
+
+    @property
+    def state(self) -> str:
+        return self._data["state"]
+
+    @property
+    def user(self) -> str:
+        return self._data["user"]["login"]
+
+    @property
+    def user_obj(self) -> User:
+        return User(self._data["user"])
+
+    @property
+    def draft(self) -> Optional[bool]:
+        if not self.is_pull_request:
+            return None
+        return self._data["draft"]
+
+    @property
+    def merged(self) -> Optional[bool]:
+        if not self.is_pull_request:
+            return None
+        return self._data["merged"]
+
+    @property
+    def allow_maintainer_edit(self) -> Optional[bool]:
+        if not self.is_pull_request:
+            return None
+        return self._data["allow_maintainer_edit"]
+
+    @property
+    def base_owner(self) -> Optional[str]:
+        if not self.is_pull_request:
+            return self._data["repository"]["owner"]
+        return self._data["base"]["repo"]["owner"]["login"]
+
+    @property
+    def base_repo(self) -> str:
+        if not self.is_pull_request:
+            return self._data["repository"]["name"]
+        return self._data["base"]["repo"]["name"]
+
+    @property
+    def base_branch(self) -> Optional[str]:
+        if not self.is_pull_request:
+            return None
+        return self._data["base"]["ref"]
 
-        if "base" in entry:
-            # a proper pull request
-            entry_id = f"{entry['base']['repo']['full_name']}#{entry['number']}"
-            is_pull_request = True
-        else:
-            # an issue without pull request details
-            entry_id = f"{entry['repository']['full_name']}#{entry['number']}"
-            is_pull_request = False
+    @property
+    def base_commit(self) -> Optional[str]:
+        if not self.is_pull_request:
+            return None
+        return self._data["base"]["sha"]
 
+    @property
+    def base_ssh_url(self) -> Optional[str]:
+        if not self.is_pull_request:
+            return None
+        return self._data["base"]["repo"]["ssh_url"]
+
+    @property
+    def head_owner(self) -> Optional[str]:
+        if not self.is_pull_request:
+            return None
+        return self._data["head"]["repo"]["owner"]["login"]
+
+    @property
+    def head_repo(self) -> Optional[str]:
+        if not self.is_pull_request:
+            return None
+        return self._data["head"]["repo"]["name"]
+
+    @property
+    def head_branch(self) -> Optional[str]:
+        if not self.is_pull_request:
+            return None
+        return self._data["head"]["ref"]
```

Comment:
Note that this reference becomes useless once the PR is merged

Reply:
Yes and no :)
What happens is this:
- `head/sha` changes and is equal to `merge_commit_sha` which is also set during merge
- `merge_base` remains set to the original `head/sha` value